### DISCUSSION TITLE
How do I specify datatypes in Timestream multi-measure record writes?

### DISCUSSION THREAD

**Discussion by nabelekt** [aws/aws-sdk-pandas#2725] [Q&A] [answered]
Hello. I am using `awswrangler.timestream.write` (https://aws-sdk-pandas.readthedocs.io/en/stable/stubs/awswrangler.timestream.write.html) to write multi-measuare records to Timestream, similar to how it's being done here: https://github.com/aws/aws-sdk-pandas/blob/main/tests/unit/test_timestream.py#L378. But, all of the values other than `time` are stored with the type `VARCHAR`. I need some to be `BIGINT`. How can I do this? I know that I can do it if I write the records using `boto3` directly, but now I am trying to do it from a Pandas dataframe. Thanks.

**Comment by nabelekt** [comment]
@jaidisido, @kukushking, or @LeonLuttenberger, can you help here? Thanks.

**Comment by LeonLuttenberger** [comment, marked as answer]
Hey,

Can you provide an example of a dataframe where the columns are returned as strings? I ran [the example from our unit tests](https://github.com/aws/aws-sdk-pandas/blob/main/tests/unit/test_timestream.py#L378), and `measure1` and `measure2` are returned as `float` values, since they're specified as measure columns.

The values that are getting converted into strings are the dimensions (`dim0` and `dim1`). Based on [the documentation](https://docs.aws.amazon.com/timestream/latest/developerguide/API_Dimension.html), it looks like they have to be strings unfortunately.

**Reply by nabelekt** [reply]
Indeed, I think all of the non-string values I was testing were being used as dimensions. Doing it differently, I have it working as expected. Thanks.
